src/idl_parser.cpp

```diff
@@ -627,12 +627,6 @@ CheckedError Parser::ParseType(Type &type) {
             "length of fixed-length array must be positive and fit to "
             "uint16_t type");
       }
-      // Check if enum arrays are used in C++ without specifying --scoped-enums
```

Comment:
Why is it better to do this in the code generator?

Reply:
1. When other langs will support arrays this check will block `flatc`.
2. The `g_only_fixed_enums` flag in idl_gen_cpp  is a more general flag. It can be set even if `--scoped-enums` is inactive (C++11 unscoped enum with explicit underlying type).